topics/03_file_handling_and_regex/hometask3.py

```diff
@@ -0,0 +1,68 @@
+###File Handling
+### 1 - Read the file and remove equal lines (if any).
+#
+with open("1.py", "r") as file:
+    lines = file.readlines()
+unic_lines = set(lines)
+with open("1.py", "w") as file:
+    file.writelines(unic_lines)
+with open("1.py", "r") as file:
+    processed_lines = file.readlines()
+    for line in processed_lines:
+        print(line)
+
+### 2 - Print out all words with length of n-characters
+n = 4   #should be specified lenght of the word
+with open("1.py", "r") as file:
+    lines = file.readlines()
+for l in lines:
+    words = l.split()
+    for word in words:
+        if len(word) == n:
+            print(word)
+
+    ##### 3 - Combine two files into a third file
+with open("1.py", "r") as file1:
+    content1= file1.read()
+with open("2.py", "r") as file2:
+    content2= file2.read()
+with open("3.py", "w+") as file3:
+    content3 = content1+content2
+print(content3)
+
+
+#### Regular Expressions
+# 1 - Define if a string contains the required characters.
+import re
+def check_string(x, y):
+    return all(re.search(a, y) for a in x)
+result1 = check_string('583', '7865serS3')
+result2 = check_string('973', '7865serS3')
+if result1:
+    print(f'Found: {result1}')
+else:
+    print("Not all characters were found")
+if result2:
+    print(result2)
+else:
+    print("Not all characters were found")
+
+# 2 - Count a number of Upper case letters in the string. E.g. '7865serS3' - 'Number of Capital letters: 1'
+
+my_pattern = '[A-Z]'
+my_string = '7865serS3IO'
+import re
+result = re.findall(my_pattern, my_string)
+capital_letters=len(result)
+print(f'number of pattern in the string {capital_letters}')
```

Comment:
Think about using assert or (at least if) - to define result - like print(int(f'{capital_letters}') == 3)